### ISSUE TITLE
[Code Health] Ensure all message/query handler return gRPC status errors

### ISSUE THREAD

**Issue by bryanchriswhite** [pokt-network/poktroll#860]
## Objective

Consistently encapsulate gRPC errors in gRPC `status.Error()` with appropriate [status codes](https://pkg.go.dev/google.golang.org/grpc@v1.67.1/codes#Code).

## Origin Document

Observations made while working on #612.

While we already often wrap errors in custom cosmos-sdk error types, we SHOULD also be wrapping any of those errors which find their way back to a gRPC client in a [gRPC status](https://pkg.go.dev/google.golang.org/grpc/status#Status) error. See: [google.golang.org/grpc/status](https://pkg.go.dev/google.golang.org/grpc/status).

## Goals

- Align with best practices.
- Improve gRPC client expectations and UX.
- Improve consistency.

## Deliverables

- [x] A PR for each module which:
  - wraps any non-status errors in all gRPC message/query handlers in an appropriately coded* gRPC status error.
  - updates status codes on existing status errors for which there is a more appropriate code*.
- [ ] A PR which adds a CLI tool which detects msg/query handler methods with non-status return errors.
  - (#929)
- [ ] A PR which runs the CLI tool in CI to mitigate regressions on the `main` branch.

_*(see: [codes.Code](https://pkg.go.dev/google.golang.org/grpc@v1.67.1/codes#Code))_

## Non-goals / Non-deliverables

- Refactoring (moving / consolidating / renaming / etc.) error variables or types.
- Refactoring message/query handler logic.

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @bryanchriswhite

**Comment by Olshansk** [comment]
@bryanchriswhite Where are we on this?

**Comment by bryanchriswhite** [comment]
Not started but expected to be low effort. I intended to do this in parallel with a higher effort issue (like #543).

**Comment by bryanchriswhite** [comment]
Closing as the fixes have all been merged. NOTE: #929 is still in progress and is what automates the mitigation of these issues going forward, once integrated into CI.
